### DISCUSSION TITLE
Can't get SD to work

### DISCUSSION THREAD

**Discussion by willjlong** [thehighnibble/thehighnibble.github.io#100] [IMSAI8080esp replica]
Processor boots but will not clear SD card read checks.  I'm pretty sure the SD socket is soldered correctly (I've run out all leads, reflowed the solder on each pin, re checked continuity to related TTB hole on PCB, checked each pin for bridging to ground).  SD card appears to have proper file system.  Anything else that might be a problem?

Thanks in advance,
Bill

**Comment by GrantMeStrength** [comment]
Have you completed the kit? I've found that some random errors went away when it was all finished.

The other options, as I'm sure you can guess, are -

* SD card reader is faulty,
* The SD cards you've tried are not compatible,
* The ESP component and/or RAM is not soldered properly..

**Reply by willjlong** [reply]
Thanks for the reply.  I really don't want to complete the project until I know I have a working processor and SD Reader.
I'm not sure about the SD socket and I don't know of an easy way to test it.  Amazon has some extenders with flat cable so that I could, at least, ring out the leads for continuity. 
The SD card came with the kit and has a readable file system.  Pretty sure it's fine.
The  ESP boots, with output over USB to my laptop running Putty.  I think that pretty much proves that the ESP and RAM are working.  Of course, there are the ESP pins associated with the SD socket which I rang through for continuity to the associated pads on the SD socket.

**Reply by willjlong** [reply]
Also no WiFi.

**Reply by GrantMeStrength** [reply]
I still think finishing it up and then trying a different SD card is the best bet, if you are happy the soldering is all good.

**Reply by willjlong** [reply]
Per your suggestion, I proceeded with the build up until the next test point.  It appears as if the SD card reader magically started working!  I'm guessing I misinterpreted the initial diagnostic log that indicated otherwise.

There are some remaining issues that will need resolution but, at least with your help, I'm making progress!!

**Reply by willjlong** [reply]
Spoke too soon.  LEDs light up but boot is back to its original log display of "The uSD tests appear to have FAILED."  Looks as if I'm still having issues reading the mSD.  I've ordered a mSD extender to help check out the mSD card socket.  Would welcome any other suggestions.

**Reply by willjlong** [reply]
Actually, am able to get an intermittent boot session that appears to read the SD before  throwing a Guru Meditation Error.  See attached log.  Is there any documentation around that explains this log?
[Boot Log 2923199031413.txt](https://github.com/thehighnibble/thehighnibble.github.io/files/12796219/Boot.Log.2923199031413.txt)

**Reply by GrantMeStrength** [reply]
Sorry you're still having issues. An SD extension won't do much, maybe better off removing the SD card component and adding a new one in its place? One more thing: I'd check the system is getting enough power, with a new USB cable and good PSU.

**Reply by nbreeden2** [reply]
As said above the USB cable needs to be a quality one. I tried several before I got my system running solid. It also needs a good quality +5V adapter. With a bad cable or poor adapter all kinds of random issue come up.

**Reply by willjlong** [reply]
Thanks for the help!  I checked the voltages on the TTL chips -  4.7 v which should be ok.  10-4 on the need for quality cables and PS.  The SD extension should allow me to ring out continuity all the way through the SD card socket to make absolutely sure.  I'll also follow the suggestion to go ahead and finish soldering up everything.  We'll see...

**Reply by nbreeden2** [reply]
What you measure on the IMSAI should be very close to what you measure on the adapter. The ESP32 can see large current spikes as the WiFi is brought up resulting in power dips. Seeing 4.7V on the IMSAI side should not be assumed to be OK. There is 0.3V being dropped in the cable. I'll say this again: Bad USB cables or adapters are one of the largest cause of issues with the IMSAI.

**Reply by willjlong** [reply]
Many thanks to John and Neil!  P:roblem was indeed the SD card reader.  Now everything is working!!

**Reply by GrantMeStrength** [reply]
Success! Now your IMSAI is ready for work.

**Comment by nbreeden2** [comment]
Glad it's working.

**Comment by pfg504** [comment]
I will try different USB cables - this is to a MAC Mini, I was able to read the SD card with a MicroSD reader on the mac directly

E (1588) sdmmc_sd: sdmmc_check_scr: send_scr returned 0xffffffff
E (1588) vfs_fat_sdmmc: sdmmc_card_init failed (0xffffffff).
E (1590) esp32_hardare: Failed to mount filesystem (ESP_FAIL).

**Reply by pfg504** [reply]
Reflowed the SD connections ... working now ...

**Comment by nbreeden2** [comment]
Awesome, glad you got it working! :)
